control/tests/freqresp_test.py

```diff
@@ -345,3 +347,26 @@ def test_phase_wrap(TF, wrap_phase, min_phase, max_phase):
     mag, phase, omega = ctrl.bode(TF, wrap_phase=wrap_phase)
     assert(min(phase) >= min_phase)
     assert(max(phase) <= max_phase)
+
+@pytest.fixture
+def pvtol_inner():
+    m = 4               # mass of aircraft
+    J = 0.0475          # inertia around pitch axis
+    r = 0.25            # distance to center of force
+    g = 9.8             # gravitational constant
+    c = 0.05            # damping factor (estimated)
+    
+    # Transfer functions for dynamics
+    Pi = ctrl.tf([r], [J, 0, 0])        # inner loop (roll)
+
+    k, a, b = 200, 2, 50                # control parameters
+    Ci = k * ctrl.tf([1, a], [1, b])    # lead compensator
+
+    return (Pi, Ci)
+
+# Regression test for Gang of 4 plots
+@nopython2
```

Comment:
Why install nose in python2 then? Just for the import?

Reply:
The problem I was having is that I need to import `image_comparison` from `matplotlib.testing.decorators` and this requires `nose` under python2 (no idea why).  And I think there was some problem with getting the comparison to work in Python2 (at some point), so I skipped the unit test.

We can probably clean it up, but since it might break Travis and since we are deprecating Python 2, I'm going to leave for now.

Reply:
I hope we can get rid of Python 2 soon.

Reply:
Thanks for the effort, but... in my experience image comparisons with matplotlib outputs are very brittle test.
Whenever matplotlib decides to change their defaults (which they have done in the past), or a font is ever so slightly different, the test will fail. It's a never ending maintenance hassle.

Probably better to mock the matplotlib interface and only check for the right interface calls.
I'm willing to help.

Reply:
To help you on your way:

def test_gangof4_pvtol(pvtol_inner):
    with patch('control.freqplot.plt') as mockplt:
        ctrl.gangof4(*pvtol_inner)
        print(mockplt.method_calls)

Reply:
Comparing with a fixed set of samples with values of arbitrary precision is equally brittle.

Reply:
The issue to test here is not about the correct value of the curves but that the plot functions produce a non-crashing and legible plot.

Reply:
@bnavigator 
Respectfully disagree, we are not testing matplotlib, we must make sure that we send the right calls to matplotlib. 

And we wouldn't compare the calls with a hard coded array, but with the data from within the gangof4 function itself, using float compares with tolerances.

I'll support your decision.

Reply:
Sorry, I still don't see a point in testing that the calls which are defined in the source code are the same calls which we retype again in the test code. There is nothing meaningful to **compare**.